src/edu/ccsu/cs505/hw2/CreditAccount.java

```diff
@@ -0,0 +1,91 @@
+package edu.ccsu.cs505.hw2;
+
+/**
+ * @author Julie Shevchenko
+ * CreditAccount class, which extends Account.
+ * Overrides setNextChain and deduct methods 
+ **/
+public class CreditAccount extends Account{
+    private double creditBalance;
+    private double creditLimit;
+    private Account chainAccount;
+
+    /** 
+     * constructor for CreditAccount
+     * takes balance, limit, and chain and sets creditBalance, creditLimit, and chainAccount variables to those values
+     * @param name accountName
+     * @param balance accountBalance
+     * @param chain chainAccount
+     **/
+    public CreditAccount(double balance, double limit, Account chain) {
+        creditBalance = balance;
+        creditLimit = limit;
+        chainAccount = chain;
+    }
+
+    /**
+     * @see setNextChain(Account nextChain) Javadoc in Account class
+     **/
+    @Override
+    public void setNextChain(Account nextChain) {
+        this.chainAccount = nextChain;
+    }
+
+    /**
+     * @see deduct(double amount) throws InsufficientFundsException Javadoc in Account class
+     **/
+    @Override
+    public void deduct(double amount) throws InsufficientFundsException {
+        double creditLeft = creditLimit - creditBalance;
+        if (amount <= creditLeft) {
+            double remaining = creditLeft - amount;
+            System.out.println("Credit card paid $" + amount + ". $" + remaining + " left before limit.");
+        } else {
+            if (chainAccount == null) {
+                throw new InsufficientFundsException("Insufficient funds.");
+            } else {
+                double remaining = amount - creditLeft;
+                System.out.println("Credit card paid $" + creditLeft + ". $" + remaining + " still needs to be paid.");
+                this.chainAccount.deduct(remaining);
```

Comment:
Logic error where if exception is thrown the balance is left in an inconsistent state - zero even though request could not be fulfilled.